File: samples/Actor/DemoActor.FSharp/BankService.fs
namespace DemoActor.FSharp.BankService

open IDemoActor.FSharp

// Annoyances:
//
// 1. Not really much, maybe just being a class?
type BankService() =
    let overdraftThreshold = -50m

    member _.Withdraw(balance: decimal, amount: decimal) =
        let updated = balance - amount

        if updated < overdraftThreshold then
            raise (OverdraftException(balance, amount))

        updated

File: examples/Actor/DemoActor/BankService.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
// 

using IDemoActorInterface;

namespace DaprDemoActor
{
    public class BankService
    {
        // Allow overdraft of up to 50 (of whatever currency).
        private readonly decimal OverdraftThreshold = -50m;

        public decimal Withdraw(decimal balance, decimal amount)
        {
            // Imagine putting some complex auditing logic here in addition to the basics.

            var updated = balance - amount;
            if (updated < OverdraftThreshold)
            {
                throw new OverdraftException(balance, amount);
            }

            return updated;
        }
    }
}
